### PR TITLE
fix: allowing traffic type dynamically for split.io

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Traffic type was being mapped from UI only, now, we have added mapping from the payload. The UI mapping will work as default traffic type if the payload field is missing.

## What is the related Linear task?

Resolves INT-2200

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3425/test-report.html

File: src/v0/destinations/splitio/data/EventConfig.json [hunk 1]
```diff
@@ -29,5 +29,17 @@
...
     },
     "required": false
+  },
+  {
+    "destKey": "trafficTypeName",
+    "sourceKeys": [
+      "traits.trafficTypeName",
+      "context.traits.trafficTypeName",
+      "properties.trafficTypeName"
+    ],
```
Comment (Gauravudia) [inline src/v0/destinations/splitio/data/EventConfig.json:39]: Why supporting all, can't we just support if from properties?

Comment (Gauravudia) [reply src/v0/destinations/splitio/data/EventConfig.json:39]: is `trafficTypeName` user trait or event property?

Comment (shrouti1507) [reply src/v0/destinations/splitio/data/EventConfig.json:39]: Traffic type can be sent in all of the calls like track, group . Some of them will not have properties. Traffic type is not a user property for sure.

**Review by Gauravudia — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]
make sure to update destination documentation

**Pull request merged**